src/app.rs

```diff
@@ -248,6 +250,7 @@ where
     ) -> Result<Response<Body>, Error> {
         // Get the host name and scheme.
         let uri = req.base_uri().path_and_query("/").build().unwrap();
+        let headers = req.headers().clone();
```

Comment:
I think this clone is necessary, thoughts?

Reply:
Does it compile without the clone? Seems like it should, but I'm not sure since the lifetime requirements of futures gets a little complicated.

Reply:
No, doesn't compile without it. The `req.into_body()` call consumes the `req` object, so the lifetimes don't work out.

Reply:
Example

![image](https://github.com/jasonwhite/rudolfs/assets/183436/36a524b6-cc6e-4ee5-8f59-741dead3b257)